tests/integration/test_ethernet.py

```diff
@@ -0,0 +1,93 @@
+import os
+import pytest
+import subprocess
+import sys
+
+VETH_NAME = "testeth"
+
+if sys.version_info >=(3, 3):
+    from unittest import mock
+else:
+    import mock
+
+def configure_network(connections):
+    parentdir = os.path.normpath(os.path.join(os.path.dirname(__file__), "../../"))
+    with mock.patch.object(
+        sys,
+        "path",
+        [parentdir, os.path.join(parentdir, "module_utils/network_lsr")] + sys.path,
+    ):
+        with mock.patch.dict(
+            "sys.modules",
+            {
+                "ansible": mock.Mock(),
+                "ansible.module_utils": __import__("module_utils"),
+            },
+        ):
+            import library.network_connections as nc
+
+    run_env = nc.RunEnvironment()
+
+    def fail_log(connections, idx, severity, msg, **kwargs):
+        if severity == nc.LogLevel.ERROR:
+            print("Error: {}".format(connections[idx]))
+            raise RuntimeError(msg)
+        else:
+            print("Log: {}".format(connections[idx]))
+
+    def do_nothing(*args, **kwargs):
+        pass
+
+    run_env._check_mode_changed = do_nothing
+    run_env.log = fail_log
+
+    cmd = nc.Cmd.create(
+        "nm",
+        run_env=run_env,
+        connections_unvalidated=connections,
+        connection_validator=nc.ArgValidator_ListConnections(),
+    )
+
+    cmd.run()
+
+
+@pytest.fixture
+def set_test_connection():
```

Comment:
```suggestion
def testnic1():
```

might make it more clear since the fixture is used like a variable in the actual test method.